### PR TITLE
skia: init at 124-unstable-2024-05-22

### PR DESCRIPTION
**Pull request by fgaz**
## Description of changes

Dependency of #325045

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by zlepper** [comment]
Result of `nixpkgs-review pr 325871` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>skia</li>
  </ul>
</details>

**Comment by qbit** [comment]
Result of `nixpkgs-review pr 325871` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>skia</li>
  </ul>
</details>

**Comment by qbit** [comment]
Result of `nixpkgs-review pr 325871` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>skia</li>
  </ul>
</details>

**Comment by qbit** [comment]
Error from darwin:
```
error: builder for '/nix/store/8xh0ayff2s5qrgvifjcgbxbccddlqp42-skia-124-unstable-2024-05-22.drv' failed with exit code 1;
       last 10 log lines:
       > FAILED: obj/src/gpu/gpu_shared.MutableTextureState.o
       > c++ -MD -MF obj/src/gpu/gpu_shared.MutableTextureState.o.d -DNDEBUG -DSK_CODEC_DECODES_BMP -DSK_CODEC_DECODES_WBMP -DSKIA_DLL -DSK_ENABLE_PRECOMPILE -DSK_ASSUME_GL=1 -DSK_GANESH -DSK_DISABLE_TRACING -DSK_USE_PERFETTO -DSK_ENABLE_API_AVAILABLE -DSK_GAMMA_APPLY_TO_A8 -DSK_ENABLE_AVX512_OPTS -DSKIA_IMPLEMENTATION=1 -I.. -Wno-attributes -ffp-contract=off -fstrict-aliasing -fPIC -fvisibility=hidden -march=armv7-a -mfpu=neon -mthumb -isysroot /nix/store/12rrl7arpahm7l708pvg1ly247sc0djg-SDKs/MacOSX11.0.sdk -target x86_64-apple-macos10.13 -O3 -I/nix/store/lha2hfsmgz1a5yrscpqaq9yfphjvi1g7-harfbuzz-icu-8.4.0-dev/include/harfbuzz -std=c++17 -fvisibility-inlines-hidden -fno-exceptions -fno-rtti -c ../src/gpu/MutableTextureState.cpp -o obj/src/gpu/gpu_shared.MutableTextureState.o
       > clang-16: warning: overriding '-mmacos-version-min=11.0' option with '-target x86_64-apple-macos10.13' [-Woverriding-t-option]
       > clang-16: error: unsupported argument 'armv7-a' to option '-march='
       > [8/1004] compile ../src/gpu/PipelineUtils.cpp
       > FAILED: obj/src/gpu/gpu_shared.PipelineUtils.o
       > c++ -MD -MF obj/src/gpu/gpu_shared.PipelineUtils.o.d -DNDEBUG -DSK_CODEC_DECODES_BMP -DSK_CODEC_DECODES_WBMP -DSKIA_DLL -DSK_ENABLE_PRECOMPILE -DSK_ASSUME_GL=1 -DSK_GANESH -DSK_DISABLE_TRACING -DSK_USE_PERFETTO -DSK_ENABLE_API_AVAILABLE -DSK_GAMMA_APPLY_TO_A8 -DSK_ENABLE_AVX512_OPTS -DSKIA_IMPLEMENTATION=1 -I.. -Wno-attributes -ffp-contract=off -fstrict-aliasing -fPIC -fvisibility=hidden -march=armv7-a -mfpu=neon -mthumb -isysroot /nix/store/12rrl7arpahm7l708pvg1ly247sc0djg-SDKs/MacOSX11.0.sdk -target x86_64-apple-macos10.13 -O3 -I/nix/store/lha2hfsmgz1a5yrscpqaq9yfphjvi1g7-harfbuzz-icu-8.4.0-dev/include/harfbuzz -std=c++17 -fvisibility-inlines-hidden -fno-exceptions -fno-rtti -c ../src/gpu/PipelineUtils.cpp -o obj/src/gpu/gpu_shared.PipelineUtils.o
       > clang-16: warning: overriding '-mmacos-version-min=11.0' option with '-target x86_64-apple-macos10.13' [-Woverriding-t-option]
       > clang-16: error: unsupported argument 'armv7-a' to option '-march='
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix-store -l /nix/store/8xh0ayff2s5qrgvifjcgbxbccddlqp42-skia-124-unstable-2024-05-22.drv'.
error: 1 dependencies of derivation '/nix/store/xvvwhcl9572fhc5q8w4dzxn61lhg2yz1-review-shell.drv' failed to build
```

`armv7-a` seems wrong? Maybe darwin support should come post-merge?

**Comment by fgaz** [comment]
darwin x86_64 also fails:

```
../include/private/base/SkTArray.h:523:49: error: no matching function for call to 'min'
    static constexpr int kMaxCapacity = SkToInt(std::min(SIZE_MAX / sizeof(T), (size_t)INT_MAX));
                                                ^~~~~~~~
```

I marked both as broken.

File: pkgs/by-name/sk/skia/package.nix [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+  preFixup = ''
+    # Some skia includes are assumed to be under an include sub directory by
+    # other includes
+    for file in $(grep -rl '#include "include/' $out/include); do
+      substituteInPlace "$file" \
+        --replace-fail '#include "include/' '#include "'
+    done
+  '';
+
```
Comment (jopejoe1) [inline pkgs/by-name/sk/skia/package.nix:129]:
```suggestion
  passthru.tests.pkg-config = testers.testMetaPkgConfig finalAttrs.finalPackage;
```

File: pkgs/by-name/sk/skia/package.nix [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  meta = {
+    description = "2D graphic library for drawing text, geometries and images";
+    homepage = "https://skia.org/";
+    license = lib.licenses.bsd3;
+    maintainers = with lib.maintainers; [ fgaz ];
+    platforms = lib.platforms.all;
```
Comment (jopejoe1) [inline pkgs/by-name/sk/skia/package.nix:135]:
```suggestion
    platforms = lib.platforms.all;
    pkgConfigModules = [  "skia" ];
```

**Review by jopejoe1 — APPROVED** [review]
Maybe also add the pkg-config module to meta.

Result of `nixpkgs-review pr 325871` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>skia</li>
  </ul>
</details>

**Review by jopejoe1 — APPROVED** [review]

**Pull request merged**